src/utils/sourcemaps/inject.rs

```diff
@@ -115,25 +115,31 @@ impl fmt::Display for InjectReport {
 /// Moreover, if a `sourceMappingURL` comment exists in the file, it is moved to the very end.
 pub fn fixup_js_file(js_contents: &mut Vec<u8>, debug_id: DebugId) -> Result<()> {
     let js_lines: Result<Vec<String>, _> = js_contents.lines().collect();
+    let mut js_lines = js_lines?;
 
-    let mut sourcemap_comment = None;
     js_contents.clear();
 
-    for line in js_lines?.into_iter() {
-        if line.starts_with("//# sourceMappingURL=") || line.starts_with("//@ sourceMappingURL=") {
-            sourcemap_comment = Some(line);
-            continue;
-        }
+    let sourcemap_comment_idx = js_lines
+        .iter()
+        .enumerate()
+        .rev()
+        .find(|(_idx, line)| {
+            line.starts_with("//# sourceMappingURL=") || line.starts_with("//@ sourceMappingURL=")
+        })
+        .map(|(idx, _)| idx);
+
+    let sourcemap_comment = sourcemap_comment_idx.map(|idx| js_lines.remove(idx));
+
+    for line in js_lines.into_iter() {
         writeln!(js_contents, "{line}")?;
     }
 
     let to_inject = CODE_SNIPPET_TEMPLATE.replace(DEBUGID_PLACEHOLDER, &debug_id.to_string());
-    writeln!(js_contents)?;
     writeln!(js_contents, "{to_inject}")?;
     writeln!(js_contents, "{DEBUGID_COMMENT_PREFIX}={debug_id}")?;
 
     if let Some(sourcemap_comment) = sourcemap_comment {
-        write!(js_contents, "{sourcemap_comment}")?;
+        writeln!(js_contents, "{sourcemap_comment}")?;
```

Comment:
just for context…
it might be the case that users are still post-processing stuff after a `sentry-cli` run, like blindly concatenating two files, in which case we would break as the first line of the concatenated file would be commented out, and the 2nd line would have a syntax error